### PR TITLE
Huawei Plugin implementation mibotiaf. added tests/regex fixes

### PR DESCRIPTION
**Pull request by nielsvanhooy**
expanded upon "Huawei Plugin implementation mibotiaf "

fixed regexes for cases where the hostname can be like "zzz-ab-9999-ax-99" matches other variants to.
looked at unittests implementation from other OS'es and added for hvrp.

i tested this in our production environment and all seems to be well :)
also handled dialog [yes/no]  (in case you want to reboot/save configuration)

### CODE REVIEW

**Review by domachad — CHANGES_REQUESTED** [review]
Hi @nielsvanhooy, thank you for your contribution!

There are just a few details you need to adjust. I left some comments below.

Thanks once again!

File: docs/changelog/undistributed/template.rst [hunk 1]
```diff
@@ -3,33 +3,27 @@ Only one changelog file per pull request. Combine these two templates where appl
 Templates
```
[the remark this answers is not in the document]
Comment (nielsvanhooy) [reply docs/changelog/undistributed/template.rst:3]: done. not sure what happened here or why that changed

File: src/unicon/plugins/hvrp/setting.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+"""
```
[the remark this answers is not in the document]
Comment (nielsvanhooy) [reply src/unicon/plugins/hvrp/setting.py]: removed.

File: src/unicon/plugins/hvrp/setting.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        self.HA_INIT_EXEC_COMMANDS = [
+            'screen-length 0 temporary',
+            'undo terminal alarm',
+            'undo terminal logging',
+            'undo terminal debugging',
+            'undo terminal monitor'
+        ]
+        self.HA_INIT_CONFIG_COMMANDS = []
+
+        self.CONSOLE_TIMEOUT = 60
```
[the remark this answers is not in the document]
Comment (nielsvanhooy) [reply src/unicon/plugins/hvrp/setting.py]: we did not find any issues with leaving this out in the settings.py file.
i can report back to this tomorrow.

File: docs/changelog/undistributed/template.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-the-iii) [reply docs/changelog/undistributed/template.rst:3]: I'm looking at the pr and I still see it as making changes to the template file. I recommend having git checkout the version of the file in dev, I think that should fix it.

**Review by d-the-iii — CHANGES_REQUESTED** [review]
Thanks for your contribution. I just need you to take a look at the template file again and make it so you're not changing it and everything should be good.

File: docs/changelog/undistributed/template.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (nielsvanhooy) [reply docs/changelog/undistributed/template.rst]: took some time to get that fixed up. but it should be ok now. i see the changed files dropped from 14 to 13 :)

**Review by domachad — APPROVED** [review]

**Review by d-the-iii — APPROVED** [review]

**Pull request merged**